cnf-certification-test/operator/suite.go

```diff
@@ -61,3 +67,34 @@ func testOperatorInstallationPhaseSucceeded(env *provider.TestEnvironment) {
 		ginkgo.Fail(fmt.Sprintf("Found %d CSVs whose phase is not %s.", n, v1alpha1.CSVPhaseSucceeded))
 	}
 }
+
+func testOperatorInstallationWithoutPrivileges(env *provider.TestEnvironment) {
+	badCsvs := []string{}
+	if len(env.Csvs) == 0 {
+		ginkgo.Skip("No CSVs to perform test, skipping.")
+	}
+
+	for _, csv := range env.Csvs {
+		clusterPermissions := csv.Spec.InstallStrategy.StrategySpec.ClusterPermissions
+		if len(clusterPermissions) == 0 {
+			logrus.Debugf("No clusterPermissions found in csv %s (ns %s)", csv.Name, csv.Namespace)
+			continue
+		}
+
+	LOOP:
+		for i := range clusterPermissions {
+			permission := clusterPermissions[i]
+			for ruleIndex := range permission.Rules {
+				if n := len(permission.Rules[ruleIndex].ResourceNames); n > 0 {
```

Comment:
```suggestion
			for _, rule := range permission.Rules {
				if n := len(rule.ResourceNames); n > 0 {
```

Reply:
I like that way, I've done it that way tons of times before... But now, working with external types (k8s ones), you don't really know the size of those objects, so each iteration means copying the full object to the iteration variable (rule, in this case). In case the object is too big, the linter will complain about this. Even if it works now because the type spec is small and the linter does not complain, you cannot guarantee that it will work forever, because in the next k8s version they might want to add new fields to the resources, making it bigger enough for the linter to complain. 
So, my way to address this now is: when iterating over arrays for external object types, I try to use iteration variables only if it's an array of pointers (like in the previous StrategySpec.ClusterPermissions loop). Otherwise, I use indexes variables like in this case.